src/main/java/uk/ac/ebi/embl/gff3tools/utils/OntologyClient.java

```diff
@@ -0,0 +1,205 @@
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package uk.ac.ebi.embl.gff3tools.utils;
+
+import java.io.InputStream;
+import java.util.*;
+import java.util.stream.Stream;
+import org.semanticweb.owlapi.apibinding.OWLManager;
+import org.semanticweb.owlapi.model.*;
+import org.semanticweb.owlapi.reasoner.OWLReasoner;
+import org.semanticweb.owlapi.reasoner.OWLReasonerFactory;
+import org.semanticweb.owlapi.reasoner.structural.StructuralReasonerFactory;
+import org.semanticweb.owlapi.search.EntitySearcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OntologyClient {
+    private static final Logger LOGGER = LoggerFactory.getLogger(OntologyClient.class);
+    private OWLOntology ontology;
+    private OWLDataFactory dataFactory;
+    private OWLReasoner reasoner;
+
+    public OntologyClient() {
+        this.dataFactory = OWLManager.createOWLOntologyManager().getOWLDataFactory();
+        loadOntology();
+    }
+
+    private void loadOntology() {
+        try (InputStream is = getClass().getClassLoader().getResourceAsStream("so.owl")) {
+            if (is == null) {
+                LOGGER.error("so.owl resource not found.");
+                return;
+            }
+            OWLOntologyManager manager = OWLManager.createOWLOntologyManager();
+            this.ontology = manager.loadOntologyFromOntologyDocument(is);
+            // Initialize the reasoner after the ontology is loaded
+            OWLReasonerFactory reasonerFactory = new StructuralReasonerFactory();
+            this.reasoner = reasonerFactory.createReasoner(ontology);
+            LOGGER.info("SO Ontology loaded successfully.");
+        } catch (OWLOntologyCreationException e) {
+            LOGGER.error("Error loading SO Ontology: " + e.getMessage(), e);
+        } catch (Exception e) {
+            LOGGER.error("Unexpected error loading SO Ontology: " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Checks if a given child ontology ID is a child of a given parent ontology ID in the loaded SO ontology.
+     *
+     * @param childOntologyId The child ontology ID (e.g., "SO:0000123").
+     * @param parentOntologyId The parent ontology ID (e.g., "SO:0000456").
+     * @return true if the child term is a child of the parent term, false otherwise.
+     */
+    public boolean isChildOf(String childOntologyId, String parentOntologyId) {
+        if (ontology == null || reasoner == null) {
+            LOGGER.warn("Ontology or reasoner not loaded. Cannot check for child relationship.");
+            return false;
+        }
+
+        OWLClass childClass = dataFactory.getOWLClass(
+                IRI.create("http://purl.obolibrary.org/obo/" + childOntologyId.replace(":", "_")));
```

Comment:
We can create a static final variable for this and other URI(s):
`static final String LIBRARY_URL = http://purl.obolibrary.org/obo/`